### PR TITLE
[REG v2] Can't have integration tests with same module name

### PR DESCRIPTION
**Pull request by leandro-lucarella-sociomantic**
```
/docker/turtle/build/devel/tmp/allunittests.d(29): Error: module main from file test/dmqfull/main.d conflicts with another module main from file test/dhtrestart/main.d
```

I think this must be consequence of 145edfcd5b050736176a1d0c1436e8c47796cf34 - all those module are now built in one go and thus must have distinct module names (== explicit module statements in this turtle case).

### CODE REVIEW

**Comment by mihails-strasuns-sociomantic** [comment]
```
Starting with v2.0.0 Makd will compile all modules from integration tests as part of regular unittest target and stop supplying `-unittest` flag when compiling actual integration tests. This may break compilation of a project so that manual adjustment is required after Makd upgrade:

1. If integration test modules don't have module statements or have one with identical module names, it will result in compilation failure. Solution is to add explicit module statements to give each file distinct module name.
2. Integration test modules containing `main` functions need to be modified to version out that `main` function via `version(UnitTest)` or unittest target will fail to link because of multiple `main`s. 
```

**Comment by leandro-lucarella-sociomantic** [comment]
Fix attached.

**Comment by leandro-lucarella-sociomantic** [comment]
Added an unrelated improvement to migration notes (unify packaging changes).

**Review by mihails-strasuns-sociomantic — COMMENTED** [review]
@gavin-norman-sociomantic please merge after english check

File: relnotes/integrationtests.migration.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+* Unit and integration tests changes
+
+  * Starting with v2.0.0 Makd will compile all modules from integration tests as part of regular unittest target and stop supplying `-unittest` flag when compiling actual integration tests. This may break compilation of a project so some manual tweaks are required:
```
Comment (gavin-norman-sociomantic) [inline relnotes/integrationtests.migration.md]: "part of the regular"

Comment (gavin-norman-sociomantic) [reply relnotes/integrationtests.migration.md]: "stop supplying the `-unittest` flag"

File: relnotes/integrationtests.migration.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+    1. If integration test modules don't have module statements or have one with identical module names, it will result in compilation failure.
```
Comment (gavin-norman-sociomantic) [inline relnotes/integrationtests.migration.md]: "or have one with identical module names" -- does this mean "or multiple modules have the same name" ? (The latter is clearer.)

File: relnotes/integrationtests.migration.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+       The solution is to add explicit module statements to give each file distinct module name.
+
+    2. As with modules containing a `main()` in the regular user code in `$(SRC)`, integration test modules containing `main()` functions need to be modified to version out that `main()` function via `version(UnitTest)`, otherwise the unittest target will fail to link because of multiple `main()`s.
+
+       Again, you could manually add these modules to `TEST_FILTER_OUT`, but it is not recommended as any unit tests present in the module with the `main()` function won't be ran this way.
```
Comment (gavin-norman-sociomantic) [inline relnotes/integrationtests.migration.md]: "won't be run"

File: relnotes/integrationtests.migration.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+  * Integration tests default location changed from `test` to `integrationtest`
+
+    The location can be set explicitly via the variable `INTEGRATIONTEST`, but you are not required to define it anymore (as it was required in v1.x.x), you just need to define it if you don't want the new default (`integrationtest`). This change was introduced to avoid symbol clashes in D2.
```
Comment (gavin-norman-sociomantic) [inline relnotes/integrationtests.migration.md]: "This change was introduced to avoid symbol clashes in D2." -- maybe explain this a little bit more? (Just from that sentence, I don't understand the reasoning.)

**Review by gavin-norman-sociomantic — COMMENTED** [review]
A few small points.

File: relnotes/integrationtests.migration.md (hunk 2 above)
Comment (leandro-lucarella-sociomantic) [reply relnotes/integrationtests.migration.md]: Yes, it seems more clear

File: relnotes/integrationtests.migration.md (hunk 4 above)
Comment (leandro-lucarella-sociomantic) [reply relnotes/integrationtests.migration.md]: Added the classical example, @mihails-strasuns-sociomantic please check if what I'm saying is correct, as I might be forgetting or misinterpreting some detail.

Comment (mihails-strasuns-sociomantic) [reply relnotes/integrationtests.migration.md]: I would rephrase it more to the point "can't directly use any symbol named `test` if you root package name is also `test`".

**Comment by leandro-lucarella-sociomantic** [comment]
Added one more clarification. Ping?

**Comment by leandro-lucarella-sociomantic** [comment]
Updated to do all the squashing.

File: relnotes/integrationtests.migration.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
+* Unit and integration tests changes
+
+  * Starting with v2.0.0 Makd will compile all modules from integration tests as part of the regular unittest target and stop supplying the `-unittest -version=UnitTest -debug=UnitTest` flags when compiling actual integration tests. This may break compilation of a project so some manual tweaks are required:
+
+    1. If integration test modules don't have module statements or multiple modules have the same name.
+
+       The solution is to add explicit module statements to give each file distinct module name.
```
Comment (joseph-wakeling-sociomantic) [inline relnotes/integrationtests.migration.md]: `a distinct module name` ... ?

File: relnotes/integrationtests.migration.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+    2. As with modules containing a `main()` in the regular user code in `$(SRC)`, integration test modules containing `main()` functions need to be modified to version out that `main()` function via `version(UnitTest)`, otherwise the unittest target will fail to link because of multiple `main()`s.
```
Comment (joseph-wakeling-sociomantic) [inline relnotes/integrationtests.migration.md]: Might be better to be entirely explicit: `version (UnitTest) {} else` ... ?

File: relnotes/pkg.migration.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+  * The `OPTS` and `ARGS` variables used when defining packages have become built-ins now, so they should be modified **in-place** (using `OPTS.update()`, `OPTS['xxx']`, and `ARGS.extend()`, `ARGS.append()` or any other functions that mutates the object).
+
+    You should **NEVER** re-bind (re-assign) these variables (`OPTS = ...`, `ARGS = ...` or even `ARGS += ...` and any other operators that re-assign the variable are forbidden).
+
+    Also, if you are using a `defaults.py` file, now just use `ìmport defaults` in `.pkg` files using it, don't explicitly import the `OPTS` and `ARGS` symbols anymore:
+
+    Change `from defaults import OPTS, ARGS` to `import defaults`.
+
+    This change helps making much compact utility functions. See changes on `FUN.desc()` for an example.
```
Comment (joseph-wakeling-sociomantic) [inline relnotes/pkg.migration.md]: `much compact` doesn't quite make sense as phrasing -- did you mean `much more compact`?

File: relnotes/pkg.migration.md [hunk 8]
```diff
@@ -0,0 +1,19 @@
+* Packaging
+
+  * The `OPTS` and `ARGS` variables used when defining packages have become built-ins now, so they should be modified **in-place** (using `OPTS.update()`, `OPTS['xxx']`, and `ARGS.extend()`, `ARGS.append()` or any other functions that mutates the object).
```
Comment (joseph-wakeling-sociomantic) [inline relnotes/pkg.migration.md:3]: Is there any reason to prefer `ARGS.extend` versus `ARGS.append`?  What's the expected difference?

**Review by joseph-wakeling-sociomantic — APPROVED** [review]
This all LGTM (it reflects well my experience of updating 4 projects to use the release candidate).  I have some minor tweak suggestions, but these needn't be considered obligatory to fix.

File: relnotes/integrationtests.migration.md (hunk 6 above)
Comment (leandro-lucarella-sociomantic) [reply relnotes/integrationtests.migration.md]: This is already described verbosely in another note, I think it will be too repetitive to include it again. But I can refer to the other note.

File: relnotes/pkg.migration.md (hunk 8 above)
Comment (leandro-lucarella-sociomantic) [reply relnotes/pkg.migration.md:3]: Append in only for one element, extend will inject a list into the list.

So:

```py
[1, 2].append(3) # [1, 2, 3]
[1, 2].append([3, 4]) # [1, 2, [3, 4]]
[1, 2].extend([3, 4]) # [1, 2, 3, 4]
```

**Comment by leandro-lucarella-sociomantic** [comment]
Updated with fixes to joe's comments. @mihails-strasuns-sociomantic  / @gavin-norman-sociomantic please merge if you don't have any further comments!
